packages/core/data_adapters/BaseAdapter.ts

```diff
@@ -244,23 +252,42 @@ export abstract class BaseFeatureDataAdapter extends BaseAdapter {
     })
   }
 
-  public async estimateRegionsStats(regions: Region[], opts?: BaseOptions) {
+  /**
+   * Calculates the "feature density" of a set of regions. The primary purpose
+   * of  this API is to alert the user if they are going to be downloading too
+   * much information, and give them a hint to zoom in to see more. The default
+   * implementation samples from the regions, downloads feature data with
+   * getFeatures, and returns an object with the form \{featureDensity:number\}
+   *
+   * Derived classes can override this to return alternative calculations for
+   * featureDensity, or they can also return an object containing a byte size
+   * calculation with the format \{bytes:number, fetchSizeLimit:number\} where
+   * fetchSizeLimit is the adapter-defined limit for what it thinks is 'too much
+   * data' (e.g. CRAM and
+   * BAM may vary on what they think too much data is)
+   */
```

Comment:
In the future if we add support for getting feature density histograms, where would that go? Would it be provided as part of these feature density stats?

Reply:
do you think we would we add a new method to the data adapter API for that? or where do you envision that going?